hack/ci/fetch_kubebuilder_release_bin.sh

```diff
@@ -0,0 +1,126 @@
+goarch=$(go env GOHOSTARCH)
+
+# Supported arch under above release version
+# See source code of standard library "build" ("go/build") for os&arch constant
+os_darwin='darwin'
+darwin_supported_arch=('amd64')
+
+# Supported arch under above release version
+# See source code of standard library "build" ("go/build") for os&arch constant
+os_linux='linux'
+linux_supported_arch=('amd64' 'arm64' 'ppc64le')
+
+kubebuilder_release_base_url='https://github.com/kubernetes-sigs/kubebuilder/releases/download'
+fetch_url_base="${kubebuilder_release_base_url}/v${kubebuilder_release_version}"
+
+
+# Fetch kubebuilder release binaries and place them under "<argument>/kubebuilder/bin/"
+function fetch_kb_bin () {
+  local dest_dir="${1}/kubebuilder"
+  local archive_file
+  local checksum_file="checksums.txt"
+  local found
+  
+  # Check whether kubebuilder release binary is provided for current environment's OS & Arch
+  if [[ ${goos} == ${os_darwin} ]]; then
+    for (( i=0; i < ${#darwin_supported_arch[@]}; i++ )); do
+      if [[ ${goarch} == ${darwin_supported_arch[${i}]} ]]; then
+        found="true"
+        break
+      fi
+    done
+  
+    if [[ -z ${found} ]]; then
+      echo "No kubebuilder supported arch under ${goos}: ${goarch}"
+      exit 1
+    fi
+  
+  elif [[ ${goos} == ${os_linux} ]]; then
+    for (( i=0; i < ${#linux_supported_arch[@]}; i++ )); do
+      if [[ ${goarch} == ${linux_supported_arch[${i}]} ]]; then
+        found="true"
+        break
+      fi
+    done
+  
+    if [[ -z ${found} ]]; then
+      echo "No kubebuilder supported arch under ${goos}: ${goarch}"
+      exit 1
+    fi
+  
+  else
+    echo "Not kubebuilder supported os: ${goos}"
+    exit 1
+  fi
+  
+  archive_file="kubebuilder_${kubebuilder_release_version}_${goos}_${goarch}.tar.gz"
+  
+  mkdir -p ${dest_dir}
+
+  # Fetch kubebuilder release binary
+  curl -fsL "${fetch_url_base}/${archive_file}" -o "/tmp/${archive_file}"
+  if [[ ${?} != 0 ]]; then
+    echo "Failed to curl kubebuilder release binary"
+    exit 1
+  fi
+
+  # Fetch checksum file
+  curl -fsL "${fetch_url_base}/${checksum_file}" -o "/tmp/${checksum_file}"
+  if [[ ${?} != 0 ]]; then
+    echo "Failed to curl checksum file"
+    exit 1
+  fi
+ 
+  # Hash & check fetched binary
+  (
+  cd /tmp
+  sha256sum -c --quiet --ignore-missing "/tmp/${checksum_file}"
+  if [[ ${?} != 0 ]]; then
+    echo "Maybe, kubebuilder release binary got broken & doesn't match hash value in ${checksum_file}"
+    exit 1
+  fi
+  )
+
+  # Unpack
+  tar -C ${dest_dir} -xzf "/tmp/${archive_file}" --strip-components=1
+  
+  # Check expected binaries exist
+  if [[ !(-x ${dest_dir}/bin/etcd && -x ${dest_dir}/bin/kubectl && -x ${dest_dir}/bin/kube-apiserver) ]]; then
+    echo "There are no expected binaries at ${dset_dir}/bin : etcd, kubectl & kube-apiserver"
+    exit 1
+  fi
+
+  # Set KUBEBUILDER_ASSETS environment variable
+  export KUBEBUILDER_ASSETS="${dest_dir}/bin"
+}
+
+fetch_kb_bin "/tmp"
```

Comment:
Nit: It might be clearer to source this script and then invoke fetch_kb_bin from the calling script, just so we know that they are going into /tmp.  But just a nit!